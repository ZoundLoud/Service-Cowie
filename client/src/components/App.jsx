import React from 'react';
import { library } from '@fortawesome/fontawesome-svg-core';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faPlayCircle, faToggleOff, faPauseCircle } from '@fortawesome/free-solid-svg-icons';
import ArtistImage from './ArtistImage';
import PlayButton from './PlayButton';
import SongInfo from './SongInfo';
import ElapsedTime from './ElapsedTime';
import SoundBar from './SoundBar';
import Comment from './Comment';
import ajax from '../../lib/ajax';

library.add(faPlayCircle);
library.add(faPauseCircle);
class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      image: [],
      songs: [],
      randomImage: [],
      playStatus: false,
    };
    this.handlePlayButton = this.handlePlayButton.bind(this);
  }

  componentDidMount() {
    this.getImage();
    this.getSongs();
    this.getRandomImage();
  }

  getImage() {
    ajax.getAllImages((err, data) => {
      if (err) {
        console.log('getImage failed', err);
      } else {
        const randomIndex = Math.floor(Math.random() * 100);
        this.setState({
          image: data[randomIndex],
        });
      }
    });
  }

  getRandomImage() {
    ajax.getAllImages((err, data) => {
      if (err) {
        console.log('getImage failed', err);
      } else {
        const img = [];
        for (let i = 0; i < 80; i += 1) {
          const randomIndex = Math.floor(Math.random() * 100);
          img.push(data[randomIndex].imgurl);
        }
        this.setState({
          randomImage: img,
        });
      }
    });
  }

  getSongs() {
    ajax.getAllSongs((err, data) => {
      if (err) {
        console.log('getSongs Failed', err);
      } else {
        const randomIndex = Math.floor(Math.random() * 100);
        this.setState({
          songs: data[randomIndex],
        });
      }
    });
  }

  handlePlayButton() {
    this.setState({
      playStatus: !this.state.playStatus,
    });
  }

  render() {
    return (
      <div>
        <div className="container">
          <ArtistImage img={this.state.image} />
          <ElapsedTime info={this.state.songs} />
          <SongInfo song={this.state.songs} />
          <PlayButton onToggle={this.handlePlayButton} play={this.state.playStatus} />
          {
            this.state.songs.decibel && this.state.songs.songlength && this.state.image.imgurl && this.state.randomImage
            && (
            <div>
              <div className="soundbar">
                <SoundBar sound={this.state.songs} play={this.state.playStatus} />
              </div>
              <div>
                <Comment img={this.state.image} random={this.state.randomImage} />
              </div>
            </div>
            )
          }
        </div>
      </div>
    );
  }
}

export default App;
