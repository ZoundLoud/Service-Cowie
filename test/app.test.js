import React from 'react';
import ReactDOM from 'react-dom';
import { configure, shallow, mount, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import App from '../client/src/components/App';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

describe('App Component', () => {
  it('should render App without throwing an error', () => {
    expect(shallow(<App />).exists()).toBe(true);
  });
  it('works', () => {
    const wrap = shallow(
      <App />,
    );
    expect(wrap.text()).toEqual('<ArtistImage /><ElapsedTime /><SongInfo /><PlayButton />');
  });
  it('Renders according to snapshot', () => {
    const app = shallow(<App />);
    expect(toJson(app)).toMatchSnapshot();
  });
  it('should render correctly in "debug" mode', () => {
    const debug = shallow(<App debug />);
  
    expect(toJson(debug)).toMatchSnapshot();
  });
  it('should mount correctly', () => {
    const wrap = mount(<App />);
    expect(toJson(wrap)).toMatchSnapshot();
  });
  it('should render correctly', () => {
    const wrap = render(<App />);
    expect(toJson(wrap)).toMatchSnapshot();
  });
});
