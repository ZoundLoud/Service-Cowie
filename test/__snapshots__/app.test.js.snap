// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Renders according to snapshot 1`] = `
<div>
  <div
    className="container"
  >
    <ArtistImage
      img={Array []}
    />
    <ElapsedTime
      info={Array []}
    />
    <SongInfo
      song={Array []}
    />
    <PlayButton
      onToggle={[Function]}
      play={false}
    />
  </div>
</div>
`;

exports[`App Component should mount correctly 1`] = `
<App>
  <div>
    <div
      className="container"
    >
      <ArtistImage
        img={Array []}
      >
        <span
          className="imageContainer"
        >
          <img
            alt=""
            className="image"
          />
        </span>
      </ArtistImage>
      <ElapsedTime
        info={Array []}
      >
        <div
          className="Time-Container"
        >
          <div
            className="Elapsed-time"
          >
            a few seconds ago
          </div>
          <button
            className="hashtag"
            type="submit"
          >
            # 
          </button>
        </div>
      </ElapsedTime>
      <SongInfo
        song={Array []}
      >
        <div
          className="Info-Container"
        >
          <div
            className="Artist-Name"
          />
          <span
            className="Song-Name"
          />
        </div>
      </SongInfo>
      <PlayButton
        onToggle={[Function]}
        play={false}
      >
        <div
          className="btnContainer"
        >
          <FontAwesomeIcon
            border={false}
            className="playBtn"
            fixedWidth={false}
            flip={null}
            icon="play-circle"
            inverse={false}
            listItem={false}
            mask={null}
            onClick={[Function]}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            style={
              Object {
                "height": "60px",
                "width": "60px",
              }
            }
            symbol={false}
            title=""
            transform={null}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-play-circle fa-w-16 playBtn"
              data-icon="play-circle"
              data-prefix="fas"
              focusable="false"
              onClick={[Function]}
              role="img"
              style={
                Object {
                  "height": "60px",
                  "width": "60px",
                }
              }
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm115.7 272l-176 101c-15.8 8.8-35.7-2.5-35.7-21V152c0-18.4 19.8-29.8 35.7-21l176 107c16.4 9.2 16.4 32.9 0 42z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </FontAwesomeIcon>
        </div>
      </PlayButton>
    </div>
  </div>
</App>
`;

exports[`App Component should render correctly 1`] = `
<div>
  <div
    class="container"
  >
    <span
      class="imageContainer"
    >
      <img
        alt=""
        class="image"
      />
    </span>
    <div
      class="Time-Container"
    >
      <div
        class="Elapsed-time"
      >
        a few seconds ago
      </div>
      <button
        class="hashtag"
        type="submit"
      >
        # 
      </button>
    </div>
    <div
      class="Info-Container"
    >
      <div
        class="Artist-Name"
      />
      <span
        class="Song-Name"
      />
    </div>
    <div
      class="btnContainer"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-play-circle fa-w-16 playBtn"
        data-icon="play-circle"
        data-prefix="fas"
        focusable="false"
        role="img"
        style="height:60px;width:60px"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm115.7 272l-176 101c-15.8 8.8-35.7-2.5-35.7-21V152c0-18.4 19.8-29.8 35.7-21l176 107c16.4 9.2 16.4 32.9 0 42z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`App Component should render correctly in "debug" mode 1`] = `
<div>
  <div
    className="container"
  >
    <ArtistImage
      img={Array []}
    />
    <ElapsedTime
      info={Array []}
    />
    <SongInfo
      song={Array []}
    />
    <PlayButton
      onToggle={[Function]}
      play={false}
    />
  </div>
</div>
`;
